src/rdpc/analysesProcessor.ts

```diff
@@ -69,6 +115,50 @@ export const analysisStream = async function* ({
   }
 };
 
+export const aggregateSpecimensByDonorId = (
+  analyses: AnalysisWithSpecimens[]
+): SpecimensByDonors => {
+  const result = analyses.reduce<SpecimensByDonors>((acc, analysis) => {
+    const donorWithSpecimens = analysis.donors.reduce<SpecimensByDonors>(
+      (_acc, donor) => {
```

Comment:
Might not be needed if this gets refactored thanks to my next comment, but, for future reference... if one code block has variables `xyz` and `_xyz` it will almost certainly be harder to follow than `resultAccumulator` and `donorsAccumulator` (for example).